### DISCUSSION TITLE
Challenges in Implementing Chat-UI with SSL Certificate on NGINX and Google Login Integration

### DISCUSSION THREAD

**Discussion by isikacer** [huggingface/chat-ui#676] [Q&A]
Hello all, need help in addressing few issues I am struggling with when securing chat-UI with  SSL certificate and login via Google.

When “Sign-in” is clicked in the Chat-UI UX

1.  Browser shows: 
      Invalid referer for POST request

2.  Inspecting Browser Console shows the following:

     POST https://mydomain.com/login 403 (Forbidden)

3.  The following is My nginx configuration:

```
server {
        server_name mydomain.com www.mydomain.com;


        location / {
        proxy_pass http://127.0.0.1:5173;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection ‘upgrade’;
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;


        }

 listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/mydomain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/mydomain.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
```

Any help the community can provide will be deeply appreciated.

**Comment by johndpope** [comment]
seems like a CORS issue. Ask chatgpt - also if you spend 5 minute learning ngrok - might be easy option to get up with ssl. 
you can port forward and will handle all the ssl for you.
